src/Core/src/Platform/Windows/NavigationViewExtensions.cs

```diff
@@ -270,5 +272,48 @@ public static void UpdateFlyoutWidth(this MauiNavigationView navigationView, IFl
 			// At some point this Template Setting is going to show up with a bump to winui
 			//handler.PlatformView.OpenPaneLength = handler.PlatformView.TemplateSettings.OpenPaneWidth;
 		}
+
+		public static async Task UpdateFlyoutIconAsync(this MauiNavigationView navigationView, IImageSource? imageSource, IImageSourceServiceProvider? provider)
+		{
+			var togglePaneButton = navigationView.TogglePaneButton;
+
+			if (togglePaneButton is null)
+				return;
+
+			var animatedIcon = togglePaneButton.GetFirstDescendant<AnimatedIcon>();
+			
+			if (animatedIcon is null)
+				return;
+
+			await animatedIcon.UpdateFlyoutIconAsync(imageSource, provider);
+		}
+		
+		public static async Task UpdateFlyoutIconAsync(this AnimatedIcon platformView, IImageSource? imageSource, IImageSourceServiceProvider? provider)
+		{
+			if (platformView is null)
+				return;			
+
+			if (provider is not null && imageSource is not null)
+			{
+				// Custom Icon
+				var service = provider.GetRequiredImageSourceService(imageSource);
+				var nativeImageSource = await service.GetImageSourceAsync(imageSource);
+
+				platformView.Source = null;
+				
+				var fallbackIconSource = new ImageIconSource { ImageSource = nativeImageSource?.Value };
+
+				platformView.Height = platformView.Width = double.NaN;
+				platformView.FallbackIconSource = fallbackIconSource;
+			}
+			else
```

Comment:
I think we still need to restore the `Source` on the platformView. When I tested this with hot reload working. The ICON was blank when returning back to the default.

Something like this work?

```C#
else if (platformView.Source is null)
{
	// Fallback to the default hamburger icon
	// https://github.com/microsoft/microsoft-ui-xaml/blob/a7183df20367bc0e2b8c825430597a5c1e6871b6/dev/NavigationView/NavigationView_rs1_themeresources.xaml#L389-L391
	var fallbackIconSource = new FontIconSource { Glyph = "&#xE700;" };
	platformView.Height = platformView.Width = 16d;
	platformView.FallbackIconSource = fallbackIconSource;
	platformView.Source = new AnimatedGlobalNavigationButtonVisualSource();
}
```

Reply:
My other thought would be to just store this Animated Icon on the `MauiNavigationView` itself. 

Then you can just swap back to it?

Reply:
> I think we still need to restore the `Source` on the platformView. When I tested this with hot reload working. The ICON was blank when returning back to the default.
> 
> Something like this work?
> 
> ```cs
> else if (platformView.Source is null)
> {
> 	// Fallback to the default hamburger icon
> 	// https://github.com/microsoft/microsoft-ui-xaml/blob/a7183df20367bc0e2b8c825430597a5c1e6871b6/dev/NavigationView/NavigationView_rs1_themeresources.xaml#L389-L391
> 	var fallbackIconSource = new FontIconSource { Glyph = "&#xE700;" };
> 	platformView.Height = platformView.Width = 16d;
> 	platformView.FallbackIconSource = fallbackIconSource;
> 	platformView.Source = new AnimatedGlobalNavigationButtonVisualSource();
> }
> ```

Right, updated and tested.